pin/gc/gc.go

```diff
@@ -69,7 +70,7 @@ func GC(ctx context.Context, bs bstore.GCBlockstore, pn pin.Pinner) (<-chan key.
 	return output, nil
 }
 
-func Descendants(ctx context.Context, ds dag.DAGService, set key.KeySet, roots []key.Key) error {
+func Descendants(ctx context.Context, ds dag.DAGService, set key.KeySet, roots []key.Key, bestEffort bool) error {
```

Comment:
i strongly dislike how this is implemented. Adding `bestEffort` here to this function ruins the whole abstraction. You can implement this "best effort" notion _outside_ of this function, just in the GC function.